wgsl/index.bs

```diff
@@ -600,28 +642,334 @@ storage_class
+to a uniform buffer variable
+or to a storage buffer variable.
+See [[#resource-layout-compatibility]].
+
+Compared to OpenGL:
+* OpenGL `std140` layout satisfies [SHORTNAME] [=uniform buffer layout=].
+* OpenGL `std430` layout satisifes [SHORTNAME] [=storage buffer layout=].
+* A type |S| with tightest offset and stride assignments satisfying [SHORTNAME] [=uniform buffer layout=] also satisfies OpenGL `std140` layout.
+* A type |S| with tightest offset and stride assignments satisfying [SHORTNAME] [=storage buffer layout=] also satisfies OpenGL `std430` layout.
+* OpenGL supports row-major matrices, but [SHORTNAME] does not.
+
+Compared to [[Vulkan1.2ext|Vulkan &sect; 15.6.4 Offset and Stride Assignment]]:
+* Vulkan *standard buffer layout* maps to [SHORTNAME] [=standard buffer layout rules=] with the following qualifications:
+    * Vulkan allows a vector to be aligned to the size of its scalar component, but [SHORTNAME] requires a more constrained alignment.
+    * The Vulkan `scalarBlockLayout` and `uniformBufferStandardLayout` features do not apply to [SHORTNAME].
+    * The Vulkan concept of *scalar alignment* does not correspond to a concept in [SHORTNAME];
+* The Vulkan *base alignment* for a type |S| corresponds to the [SHORTNAME] alignment requirement for |S| in the `storage` storage class: |Align|(|S|,`storage`).
+* The Vulkan *extended alignment* for a type |S| corresponds to the [SHORTNAME] alignment requirement for |S| in the `uniform` storage class: |Align|(|S|,`uniform`).
+* The Vulkan concept of *improperly straddle* is not permitted in [SHORTNAME], because [SHORTNAME] requires vectors to be aligned to their whole size.
+* Vulkan supports non-32-bit scalar types and vector types with non-32-bit components, but [SHORTNAME] does not.
+* Vulkan supports row-major matrices, but [SHORTNAME] does not.
+* Vulkan allows offsets to be non-monotonic, but [SHORTNAME] does not.
+
+#### Internal Layout of Values #### {#internal-value-layout}
+
+This section describes how the internals of a value are placed in the byte locations
+of a buffer, given an assumed placement of the overall value.
+These layouts depend on the value's type, the storage class of the buffer,
+the [=stride=] attribute on array types, and the [=offset=] attribute on structure type
+members.
+
+When a value |V| of type [=u32=] or [=i32=] is placed at byte offset |k| of a host-shared buffer, then:
+   * Byte |k| contains bits 0 through 7 of |V|
+   * Byte |k|+1 contains bits 8 through 15 of |V|
+   * Byte |k|+2 contains bits 16 through 23 of |V|
+   * Byte |k|+3 contains bits 24 through 31 of |V|
+
+Note: Recall that [=i32=] uses twos-complement representation, so the sign bit
+is in bit position 31.
+
+A value |V| of type [=f32=] is represented in IEEE 754 binary32 format.
+It has one sign bit, 8 exponent bits, and 23 fraction bits.
+When |V| is placed at byte offset |k| of host-shared buffer, then:
+   * Byte |k| contains bits 0 through 7 of the fraction.
```

Comment:
I wonder if this can be nicer visually encoded, as a follow up. For a crazy idea, it could be an 1x32 table, where a bunch of columns are merged.

Reply:
Definitely. I was going to look at that as an editorial improvement.    I'd have to learn something about drawing diagrams.